### PR TITLE
Gargantua Blood Rush now removes bolas

### PR DESCRIPTION
**Pull request by chuga-git**
## What Does This PR Do

- Slightly refactors `/mob/living/carbon/proc/uncuff()`
  - Renames the proc to `clear_restraints()`
  - Splits clearing handcuffs and legcuffs into separate procs.
  - These procs now have a return value of either `TRUE` or `FALSE` depending on whether or not the respective item existed and was removed.
  - All three procs now have the argument `show_message` (defaults to`FALSE`), which when `TRUE` will display a `visible_message` about the restraint being removed.

Gameplay changes:
- Blood Rush cannot be activated when the vampire is stamcritted, stunned, or buckled. This means that you will not waste blood when you can't benefit from the speed.
- Blood Rush now clears knockdowns, instantly stands the vampire up, and removes any legcuffs (bolas, beartraps).

## Why It's Good For The Game

The removal of Shadowstep left Gargantua as the only Vampire subclass without some kind of teleport escape ability. Stomp and Blood Rush are supposed to take Shadowstep's place, but they have a fatal flaw: you can't use them while you're bola'd, unlike a teleport. This PR brings Gargantua up to a rough parity with other subclasses.
## Testing

Made sure the refactor didn't break freedoms or cuff removal and that the ability didn't do anything weird.
## Changelog
:cl:
tweak: The Gargantua Vampire Blood Rush ability now removes bolas and clears knockdowns when the user is not buckled.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/carbon/carbon_procs.dm [hunk 1]
```diff
@@ -1018,35 +1018,91 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
...
+ * * show_message - if TRUE, will display a visible message when restraints are removed. FALSE by default.
+ *
+ * Returns:
+ * * TRUE if handcuffs or legcuffs were removed, FALSE otherwise
+ */
+/mob/living/carbon/proc/clear_restraints(show_message = FALSE)
+	var/cleared_hands = clear_handcuffs(show_message)
+	var/cleared_legs = clear_legcuffs(show_message)
+
+	return cleared_hands || cleared_legs
```
Comment (Contrabang) [inline code/modules/mob/living/carbon/carbon_procs.dm:1034]:
```suggestion
	. = clear_handcuffs(show_message)
	. |= clear_legcuffs(show_message)
```

**Comment by 1080pCat** [comment]
epic, i've been wanting garg to get some kind of freedoms ability, or bloodrush greatly lowering the resist timer. 
hopefully this is merged

**Review by DGamerL — APPROVED** [review]

File: code/modules/mob/living/carbon/carbon_procs.dm [hunk 2]
```diff
@@ -1018,35 +1018,89 @@ GLOBAL_LIST_INIT(ventcrawl_machinery, list(/obj/machinery/atmospherics/unary/ven
...
+ *
+ * Returns:
+ * * TRUE if legcuffs existed and were successfully removed, FALSE otherwise
+ */
+/mob/living/carbon/proc/clear_legcuffs(show_message = FALSE)
+	if(!legcuffed)
+		return FALSE
+
+	var/obj/item/W = legcuffed
+	if(isnull(W))
```
Comment (lewcc) [inline code/modules/mob/living/carbon/carbon_procs.dm:1084]: Since we check if(!legcuffed) just above, isn't this a pointless check?

**Comment by lewcc** [comment]
my ass forgot I was reviewing an already merged PR....

**Pull request merged**